src/functional/ffront/field_operator_ast.py

```diff
@@ -20,48 +20,108 @@
 
 import eve
 from eve import Node
-from eve.type_definitions import SymbolRef
+from eve.type_definitions import SourceLocation, StrEnum, SymbolRef
+
+
+class LocatedNode(Node):
+    location: SourceLocation
 
 
 class SymbolName(eve.traits.SymbolName):
     regex = re.compile(r"^[a-zA-Z_][\w$]*$")
 
 
-class Sym(Node):
+class Sym(LocatedNode):
     id: SymbolName  # noqa: A003
 
 
-class Expr(Node):
+class Expr(LocatedNode):
     ...
 
 
-class SymExpr(Expr):
-    id: SymbolName  # noqa: A003
-    expr: Expr
-
-
 class SymRef(Expr):
     id: SymbolRef  # noqa: A003
 
 
-class Return(Expr):
-    value: Expr
-
-
 class Name(Expr):
     id: SymbolName  # noqa: A003
 
 
 class Subscript(Expr):
-    expr: Expr
+    value: Expr
     index: int
 
 
 class Tuple(Expr):
     elts: list[Expr]
 
 
-class FieldOperator(Node):
+class UnaryOperator(StrEnum):
+    UADD = "plus"
+    USUB = "minus"
+    NOT = "not_"
+
+
+class UnaryOp(Expr):
+    op: UnaryOperator
+    operand: Expr
+
+
+class BinaryOperator(StrEnum):
+    ADD = "plus"
+    SUB = "minus"
+    MULT = "multiplies"
+    DIV = "divides"
+
+
+class BinOp(Expr):
+    op: BinaryOperator
+    left: Expr
+    right: Expr
+
+
+class CompareOperator(StrEnum):
+    GT = "greater"
+    LT = "less"
+    EQ = "eq"
+
+
+class Compare(Expr):
+    op: CompareOperator
+    left: Expr
+    right: Expr
+
+
+class BoolOperator(StrEnum):
+    AND = "and_"
+    OR = "or_"
+
+
+class BoolOp(Expr):
+    op: BoolOperator
+    left: Expr
+    right: Expr
```

Comment:
I don't think these nodes are needed until we introduce scalars. To be coherent with Python semantics and NumPy API, the logical `and` and `or` between fields should be represented using the `&` and `|` operators.

Reply:
Should we remove them for now?

Reply:
Yes, I would remove them or comment them out.